R/activate.R

```diff
@@ -78,7 +78,7 @@ renv_activate_impl <- function(project,
     renv_imbue_self(project)
 
   # restart session if requested
-  if (restart && !renv_tests_running())
+  if (restart && !is_testing())
```

Comment:
We really should pick one of these and use it everywhere (although I guess they do mean slightly different things).

Reply:
We need to maintain the distinction because there's two scenarios relevant for testing:

1. Running `renv`'s own test suite;
2. Running another package's test suite, which uses `renv`.

This popped up in `jetpack`, where some of the assumptions we made for `renv`'s local tests weren't valid there.